src/CustomerProfile.js

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useState } from "react";
+
+export default function CustomerProfile({ id }) {
+  // console.log("CustomerProfile props:", id);
+  const [clientData, setClientData] = useState(null);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const fetchClientData = async () => {
+    try {
+      setError(false);
+      setLoading(true);
+      const response = await fetch(
+        `https://cyf-react.glitch.me/customers/${id}`
+      );
+      // console.log("fetchClientData response", response);
+      if (!response.ok) {
+        throw Error();
+      }
+      const data = await response.json();
+      // console.log("fetchClientData data", data);
+      setClientData(data);
+    } catch (error) {
+      setError(true);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    if (id) {
+      fetchClientData();
+    }
+  }, [id]);
+
+  return (
```

Comment:
Instead of deciding the rendering within `return`, do you thiink moving the condition out and returning accordlingly is more readable?

```
if(loading) {
  retrun (
    <div>Loading...</div>
  )
}

if(error) {
  return (
  <div>Error</div>
  )
}

return (
  <>
    ...
  </>
 )

```